pkg/controller/s3/bucket/lifecycleConfig.go

```diff
@@ -137,10 +137,12 @@ func GenerateLifecycleRules(in []v1beta1.LifecycleRule) []awss3.LifecycleRule {
 			rule.Transitions = make([]awss3.Transition, len(local.Transitions))
 			for tIndex, transition := range local.Transitions {
 				rule.Transitions[tIndex] = awss3.Transition{
-					Date:         &transition.Date.Time,
```

Comment:
You can test that this does indeed panic by removing the date from the input of the test for `GenerateLifecycleConfig()` and then test that it works correctly now by adding it back and seeing `Date` populated